client/src/components/common/boost/BoostCalibration.tsx

```diff
@@ -0,0 +1,86 @@
+import React , {useCallback, useState} from 'react';
+import {Button, Card, InputGroup, FormControl} from 'react-bootstrap';
+import {setCalibration as sendCalibrationValue, resetCalibration} from 'api/common/powerModel';
+// import {useSensorData} from 'api/v2/sensors/data';
+
+/**
+ * Boost distance claibration input form
+ * 
+ * @returns {React.Component} Component
+ */
+function BoostCalibrationInput() {
+    const [calibrationValue, setCalib] = useState('');
+
+    const handleCalibrationChange = useCallback(
+        (event) => {
+        setCalib(event.target.value);
+        },
+        [setCalib],
+    );
+
+    const onSet = useCallback(
+        () => {
+          const calibValue: number = parseInt(calibrationValue, 10);
+          if (Number.isNaN(calibValue)) {
+            alert("Please enter an integer value only!");
+          }
+          else {
+            console.log(calibValue);
+            sendCalibrationValue(parseInt(calibrationValue, 10));
+          }
+          setCalib('');
+        },
+        [calibrationValue, setCalib],
+    );
+
+    const handleKeyPressed = useCallback(
+        (event) => { if (event.key === 'Enter') onSet(); },
+        [onSet]
+    );
+
+    return (
+    <InputGroup className="mb-1" >
+        <FormControl
+            onChange={handleCalibrationChange}
+            onKeyPress={handleKeyPressed}
+            placeholder="Calibrate distance..."
+            value={calibrationValue}
+        />
+        <InputGroup.Append className="l5">
+            <Button className="mr-2" variant="outline-secondary" onClick={onSet}>Set</Button>
+        </InputGroup.Append>
+        
+        <Button variant="outline-danger" onClick={resetCalibration}>Reset</Button>
+    </InputGroup>
+    );
+}
+
+/**
+ * Boost calibration card, displays the calibrated distance and the 
+ * distance travelled by the bike
+ * 
+ * @returns {React.Component} Component
+ */
+export default function BoostCalibration() {
+    // const GetDistTravelled = () => {
+    //     return useSensorData("reed_distance").data;
+    // };
+
+    return (
+        <Card >
+            <Card.Body>
+                <Card.Title>Calibration</Card.Title>
+                <Card.Text>
+                    BOOST may use a distance different to the bike&apos;s travelled distance for the purposes of generating power plan data.
```

Comment:
To avoid the awkward apostrophe thing, we can do a sneaky thing in React. Rather than putting the text as plain html which doesn't like fancy symbols, you can give it a JS string instead: `<Card.Text>{"I'm a JavaScript string"}</Card.Text>`

Reply:
Wow, that's actually so smart 😂.

Reply:
when I tried implementing it, `yarn` complains about unnecessary use of curly brackets

Reply:
That's rather annoying... Happy for you to leave it as is then